tests/transports/test_all_plugins.py

```diff
@@ -365,278 +297,244 @@ def test_dir_reading_permissions(custom_transport):
+
+        remote_file_name = 'file_remote.txt'
 
+        # here use full path in src and dst
+        local_file_abs_path = str(local_dir / directory / local_file_name)
+        retrieved_file_abs_path = str(local_dir / directory / retrieved_file_name)
+        remote_file_abs_path = str(remote_dir / directory / remote_file_name)
 
-def test_put_and_get_file(custom_transport):
+        text = 'Viva Verdi\n'
+        with open(local_file_abs_path, 'w', encoding='utf8') as fhandle:
+            fhandle.write(text)
+
+        transport.put(local_file_abs_path, remote_file_abs_path)
+        transport.get(remote_file_abs_path, retrieved_file_abs_path)
+
+        list_of_files = transport.listdir(str(remote_dir / directory))
+        # it is False because local_file_name has the full path,
+        # while list_of_files has not
+        assert local_file_name not in list_of_files
+        assert remote_file_name in list_of_files
+        assert retrieved_file_name not in list_of_files
+
+
+def test_putfile_and_getfile(custom_transport, tmp_path_factory):
     """Test putting and getting files."""
-    local_dir = os.path.join('/', 'tmp')
-    remote_dir = local_dir
+    local_dir = tmp_path_factory.mktemp('local')
+    remote_dir = tmp_path_factory.mktemp('remote')
+
     directory = 'tmp_try'
 
     with custom_transport as transport:
-        transport.chdir(remote_dir)
-        while transport.isdir(directory):
-            # I append a random letter/number until it is unique
-            directory += random.choice(string.ascii_uppercase + string.digits)
+        (local_dir / directory).mkdir()
+        transport.mkdir(str(remote_dir / directory))
 
-        transport.mkdir(directory)
-        transport.chdir(directory)
+        local_file_name = 'file.txt'
+        retrieved_file_name = 'file_retrieved.txt'
 
-        local_file_name = os.path.join(local_dir, directory, 'file.txt')
         remote_file_name = 'file_remote.txt'
-        retrieved_file_name = os.path.join(local_dir, directory, 'file_retrieved.txt')
+
+        # here use full path in src and dst
+        local_file_abs_path = str(local_dir / directory / local_file_name)
+        retrieved_file_abs_path = str(local_dir / directory / retrieved_file_name)
+        remote_file_abs_path = str(remote_dir / directory / remote_file_name)
 
         text = 'Viva Verdi\n'
-        with open(local_file_name, 'w', encoding='utf8') as fhandle:
+        with open(local_file_abs_path, 'w', encoding='utf8') as fhandle:
             fhandle.write(text)
 
-        # here use full path in src and dst
-        transport.put(local_file_name, remote_file_name)
-        transport.get(remote_file_name, retrieved_file_name)
-        transport.putfile(local_file_name, remote_file_name)
-        transport.getfile(remote_file_name, retrieved_file_name)
+        transport.putfile(local_file_abs_path, remote_file_abs_path)
+        transport.getfile(remote_file_abs_path, retrieved_file_abs_path)
```

Comment:
I know it wasn't done properly, but perhaps we leave it like this so that both `getfile` & `get` somehow would get tested?